src/Symfony/Component/Translation/Dumper/YamlFileDumper.php

```diff
@@ -36,4 +84,73 @@ protected function getExtension()
     {
         return 'yml';
     }
+
+    private function expandToTree(array $messages)
+    {
+        $tree = array();
+
+        foreach ($messages as $id => $value) {
+            $this->expandElement($tree, explode('.', $id), $value);
```

Comment:
@stof IMO we need validate id, bellow an example that we need to avoid it:
```php
array(
    'foo3.bar' => 'test7',
    'foo3.' => 'test7',
));
```

```yaml
foo3:
    bar: test7
    '': test7
```